src/ast_types.py

```diff
@@ -196,6 +196,20 @@ def decayed_expr_type(expr: ca.Node, typemap: TypeMap) -> SimpleType:
     return pointer_decay(expr_type(expr, typemap), typemap)
 
 
+def resolve_struct_def(struct: ca.TypeDecl, typemap: TypeMap) -> Optional[ca.Struct]:
+    """Resolve struct definition for a given type.
+    """
+    if isinstance(struct.type, ca.IdentifierType):
+        idType = struct.type.names[0]
+        if idType in typemap.typedefs:
+            return resolve_struct_def(typemap.typedefs[idType], typemap)
```

Comment:
This looks like it shouldn't type-check: `typemap.typedefs[...]` is a `Type`, while `resolve_struct_def` accepts a ca.TypeDecl. Did you run `mypy`?

Slight shortening (though will need some isinstance TypeDecl check to work with mypy):
```suggestion
    struct = resolve_typedefs(struct, typemap)
    if isinstance(struct.type, ca.IdentifierType):
```

Reply:
Done + mypy fixes